src/context/WalletProvider/config.ts

```diff
@@ -89,7 +98,26 @@ export interface SupportedWalletInfo {
   connectedWalletMenuInitialPath?: WalletConnectedRoutes
   connectedMenuComponent?: React.ComponentType<any>
 }
-export const SUPPORTED_WALLETS: Record<KeyManager, SupportedWalletInfo> = {
+
+export type SupportedWalletInfoByKeyManager = {
+  [KeyManager.Coinbase]: SupportedWalletInfo<typeof CoinbaseAdapter>
+  // Native, Mobile, and Demo wallets are all native wallets
+  [KeyManager.Native]: SupportedWalletInfo<typeof NativeAdapter>
+  [KeyManager.Mobile]: SupportedWalletInfo<typeof NativeAdapter>
+  [KeyManager.Demo]: SupportedWalletInfo<typeof NativeAdapter>
+  // TODO(gomes): export WebUSBKeepKeyAdapter as a type in hdwallet, not a declare const
```

Comment:
Most likely the root cause of all the ts-ignore across this PR

Reply:
Keeping it as-is for now, the types refactor required to make this work would be a lot more complex, rage quitted after ~2 hours